### ISSUE TITLE
Self-hosted, localhost onyl test image

### ISSUE THREAD

**Issue by wiwwo** [mastodon/mastodon#18293]
### Pitch

TL;DR: I'd love to have a completely isolated, local only mastodon instance without having to deal with SMTP and/or (more importantly) nginx/SSLas in steps
1. `git clone whatever`
2. `cd whatever`
3. `docker-compose up -d`
4. open `localhost:3000` in browser.

---
Longer, boring:
I am willing to host a mastodon instance, but I want to play with administration first.
For that, I want to startup a local, completely isolated, docker instance.
Even not being a completely beginner, I kept failing in that, mainly because:
* i don't want and need to configure any SMTP mailing whatsoever
and, more importantly,
* i do not want to run nginx on my local machine, let alone not to have to get a SSL certificate.

I have my entri in `/etc/hosts` that has `127.0.0.1 example.co` (without `m`, i know), and i'd like to have s mastodon instance without having nothing more to do.

Any way I can get that done?

### Motivation

Easy learn hot to administer, moderate and generally run a mastodon instance, without having to run a VPS just to have a letsencrypt key

**Comment by trwnh** [comment]
https://docs.joinmastodon.org/dev/setup/ should be helpful here -- you don't have to setup nginx certbot or smtp, you can just install foreman and use that instead

**Comment by Frederic-Zhou** [comment]
@gigatexal I also have the same confusion, Is there any good news?

**Comment by Frederic-Zhou** [comment]
I solved this problem by using an nginx container, so there is no need to install nginx on the host machine.

1. create local key and cert files
2. add nginx to docker-compose.yaml

docker-compose.yaml
```
  nginx:
    image: nginx:latest
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf
      - ./certs:/etc/nginx/certs
    networks:
      - external_network
      - internal_network
    ports:
      - "80:80"
      - "443:443"
    depends_on:
      - web

```

3. set nginx.conf
```
events {
  worker_connections 1024;
}

http {
  server {
    listen 80;
    server_name localhost;
    return 301 https://$host$request_uri;
  }

  server {
    listen 443 ssl;
    server_name localhost;

    ssl_certificate /etc/nginx/certs/localhost.crt;
    ssl_certificate_key /etc/nginx/certs/localhost.key;

    location / {
      proxy_pass http://web:3000;
      proxy_set_header Host $host;
      proxy_set_header X-Real-IP $remote_addr;
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header X-Forwarded-Proto $scheme;
    }
  }
}

```

4. docker-compose down && docker-compose up -d
5. visit https://localhost

> set LOCAL_DOMAIN=localhost in .env.production
